server/text_generation_server/utils/weights.py

```diff
@@ -163,8 +200,13 @@ def get_tensor(self, tensor_name: str, to_device=True):
         tensor = f.get_tensor(tensor_name)
         # Special case for gptq which shouldn't convert
         # u4 which are disguised as int32. Exl2 uses int16
-        # as well.
-        if tensor.dtype not in [torch.int16, torch.int32, torch.int64]:
+        # as well. FP8 uses torch.float8_e4m3fn
+        if tensor.dtype not in [
+            torch.float8_e4m3fn,
```

Comment:
This would only be needed when we want to support FP8 checkpoints I think? Same below.

Reply:
Yes that's one of the goal of this PR too.